tests/_dmctl_tools/check_master_online_http.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/pingcap/dm/tests/utils"
+
+	toolutils "github.com/pingcap/tidb-tools/pkg/utils"
+)
+
+// use show-ddl-locks request to test DM-master is online
+func main() {
+	addr := os.Args[1]
+	sslCA := ""
+	sslCert := ""
+	sslKey := ""
+
+	//secureOpt := grpc.WithInsecure()
+	if len(os.Args) == 5 {
+		sslCA = os.Args[2]
+		sslCert = os.Args[3]
+		sslKey = os.Args[4]
+	}
+
+	tls, err := toolutils.NewTLS(sslCA, sslCert, sslKey, "", nil)
+	if err != nil {
+		utils.ExitWithError(err)
+	}
+
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	tlsCfg := tls.TLSConfig()
+	tlsCfg.InsecureSkipVerify = true
```

Comment:
I guess it will panic when `tlsCfg` is `nil`, that happens when doesn't provide `sslCA, ...` in `os.Args` 🤔

Reply:
yes, I fix it in https://github.com/pingcap/dm/pull/569/commits/0f36bb0d7a9892bc7bc68bc6a847f618bcdab660